src/Storages/StorageMerge.cpp

```diff
@@ -604,142 +605,164 @@ std::vector<ReadFromMerge::ChildPlan> ReadFromMerge::createChildrenPlans(SelectQ
         query_info_.input_order_info = input_sorting_info;
     }
 
+    auto logger = getLogger("StorageMerge");
```

Comment:
BTW maybe it make sense to make it a member already (we can include a Merge table name as well there, as for MergeTree)?

Reply:
Let's do this in follow-up PR, this PR is too green, it would be a shame not to merge :)